pkg/webhooks/mutator/namespace_mutator.go

```diff
@@ -7,23 +7,19 @@ import (
 	admissionv1 "k8s.io/api/admission/v1"
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
-
-	"github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
-	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
 )
 
 const (
 	ignoreOperationMessage   = "ignoring other operations"
-	admittingDeletionMessage = "the namespace doesn't contain HyperConverged CR, admitting its deletion"
-	deniedDeletionMessage    = "HyperConverged CR is still present, please remove it before deleting the containing namespace"
+	admittingDeletionMessage = "the hcoNamespace doesn't contain HyperConverged CR, admitting its deletion"
+	deniedDeletionMessage    = "HyperConverged CR is still present, please remove it before deleting the containing hcoNamespace"
```

Comment:
**IMPORTANT**
you changed here the error message, replacing `namespace` to `hcoNamespace`.
the delete namespace test checks the exact error message, thus you need to update it also there:
https://github.com/kubevirt/hyperconverged-cluster-operator/blob/43945370a6322b92404a817d6cc8e99292552ab7/hack/test_delete_ns.sh#L41

Reply:
Thanks a lot!!!
For the record, I think it's a bad idea to expect to the full error message. As we saw, it leads to surprising errors and enforces the developer to duplicate the error code in two places. I think it should be fixed in future PRs :+1:

Reply:
it's just because you've decided to change `namespace` to `hcoNamespace` :)
this error message was the same for years